### PR TITLE
[Kogito-8223]Introduce a docker registry for the Container builder tests with Docker and Podman

### PR DESCRIPTION
**Pull request by desmax74**
See:https://issues.redhat.com/browse/KOGITO-8223

Signed-off-by: desmax74 <mdessi@redhat.com>

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

### CODE REVIEW

File: registry/test/docker.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	}
+	for _, imagex := range imageList {
+		if imagex.RepoTags[0] == REGISTRY || strings.HasPrefix(imagex.RepoDigests[0], REGISTRY) {
+			return true
+		}
+	}
+	return false
+}
+
+func (d Docker) RemoveRegistriContainerAndImage() {
```
Comment (ricardozanini) [inline registry/test/docker.go]:
```suggestion
func (d Docker) RemoveRegistryContainerAndImage() {
```

File: registry/test/docker_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestRegistryWithDocker(t *testing.T) {
+
+	d := Docker{}
+	defer d.RemoveRegistriContainerAndImage()
```
Comment (ricardozanini) [inline registry/test/docker_test.go]:
```suggestion
	defer d.RemoveRegistryContainerAndImage()
```

File: registry/test/podman.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func (p Podman) RemoveRegistriContainerAndImage() {
```
Comment (ricardozanini) [inline registry/test/podman.go]:
```suggestion
func (p Podman) RemoveRegistryContainerAndImage() {
```

File: registry/test/podman_test.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRegistryWithPodman(t *testing.T) {
+
+	p := Podman{}
+	defer p.RemoveRegistriContainerAndImage()
```
Comment (ricardozanini) [inline registry/test/podman_test.go]:
```suggestion
	defer p.RemoveRegistryContainerAndImage()
```

File: registry/test/docker.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+)
+
+func (d Docker) StartRegistry() bool {
+	cli, err := client.NewClientWithOpts(client.FromEnv)
```
Comment (ricardozanini) [inline registry/test/docker.go:30]: Isn't this the actual local docker registry?

File: registry/test/registry.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+const REGISTRY = "registry"
+const REGISTRY_FULL = "docker.io/library/registry"
+
+type Registry interface {
+	StartRegistry()
+	StopRegistry()
+}
+
+type Docker struct{}
```
Comment (spolti) [inline registry/test/registry.go]: Not sure if it make sense, why having a interface with empty struct?

**Review by radtriste — APPROVED** [review]
May I ask what is the use case here ?
I cannot see any description in issue or PR ...

File: registry/test/docker.go (hunk 5 above)
Comment (desmax74) [reply registry/test/docker.go:30]: In this first impl yes, it pull and start a docker registry locally to be used in our builds to have something easily to check destroy and recreate

File: registry/test/registry.go (hunk 6 above)
Comment (desmax74) [reply registry/test/registry.go]: The functions on Docker and Podman use the same signatures but the difference is the struct on which are applied, now added the connection to customize the connection when needed

**Comment by desmax74** [comment]
@radtriste added the description on the related Jira

**Review by ricardozanini — APPROVED** [review]

**Comment by radtriste** [comment]
> May I ask what is the use case here ? I cannot see any description in issue or PR ...

Thank you !

**Review by davidesalerno — APPROVED** [review]

**Pull request merged**